ystia/yorc-a4c-plugin
# Feature/gh 82 ssl config

# Pull Request description

When adding a Yorc Orchestator to Alien4Cloud, instead of requiring to add a certificate authority to a trustore, then add a client certificate to a keystore, then restart Alien4cloud with java options set to specify paths/passwords to trustore and keystore,
added configuration parameters in the Alien4Cloud Yorc plugin, so that the user can provide when configuring an Orchestrator connection :
  * a CA certificate
  * a Client key
  * a Client Certificate

## Description of the change

### What I did

In alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/ProviderConfig.java:
Added new config parameters caCertificate, clientCertificate, clientKey.

In alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/rest/RestClient.java
Using a custom trustore/keystore populated with these certificates to establish a secure connection with the orchestrator.

In in alien4cloud-yorc-plugin/pom.xml:
Fixed the httpclient version  to be in sync with the version used in Alien4Cloud runtime.
Added a dependency on bouncycastle to use classes allowing to get a private key algorithm.

### How to verify it

These changes can be verified following the `how to verify` section of the pull request on secure setup bootstrap: https://github.com/ystia/yorc/pull/312

### Description for the changelog

Add SSL configuration parameters to connect to a secure Yorc Server ([GH-82](https://github.com/ystia/yorc-a4c-plugin/issues/82))

## Applicable Issues

https://github.com/ystia/yorc-a4c-plugin/issues/82
